src/SFA.DAS.AANHub.Application/Attendances/Commands/PutAttendance/PutAttendanceCommandHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Common;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using System.Text.Json;
+
+namespace SFA.DAS.AANHub.Application.Attendances.Commands.PutAttendance;
+
+public class PutAttendanceCommandHandler : IRequestHandler<PutAttendanceCommand, ValidatedResponse<PutCommandResult>>
+{
+    private readonly IAanDataContext _aanDataContext;
+    private readonly IAuditWriteRepository _auditWriteRepository;
+    private readonly IAttendancesWriteRepository _attendancesWriteRepository;
+                                       
+    public PutAttendanceCommandHandler(IAanDataContext aanDataContext,
+                                       IAuditWriteRepository auditWriteRepository,
+                                       IAttendancesWriteRepository attendancesWriteRepository)
+    {
+        _aanDataContext = aanDataContext;
+        _auditWriteRepository = auditWriteRepository;
+        _attendancesWriteRepository = attendancesWriteRepository;
+    }
+
+    public async Task<ValidatedResponse<PutCommandResult>> Handle(PutAttendanceCommand command, CancellationToken cancellationToken)
+    {
+        var existingAttendance = await _attendancesWriteRepository.GetAttendance(command.CalendarEventId, command.RequestedByMemberId);
+
+        if (existingAttendance != null)
```

Comment:
I think this can be simplified a bit by checking exit conditions first. 
```
if ( (existingAttendance == null && !request,IsActive) || (existingAttendance !=null && request.IsActive == existingAttendance.IsActive) ) return new ValidatedResponse(Unit.Value);

return existingAttendance == null ? CreateAttendance() : UpdateAttendance();
```

Reply:
Used switch expression as discussed